docs/SPEC.md

```diff
@@ -2571,23 +2571,80 @@ from(bucket: "telegraf/autogen")
 
 #### Difference
 
-Difference computes the difference between subsequent non null records.
+Difference computes the difference between subsequent records.  
+Every user-specified column of numeric type is subtracted while others are kept intact.
 
 Difference has the following properties:
 
 * `nonNegative` bool
     nonNegative indicates if the derivative is allowed to be negative.
-    If a value is encountered which is less than the previous value then it is assumed the previous value should have been a zero.
+    If a value is encountered which is less than the previous value then the result is null.
 * `columns` list strings
     columns is a list of columns on which to compute the difference.
     Defaults to `["_value"]`.
 
+Rules for subtracting values for numeric types:
+
+ - If some value is the first ever seen, then there is nothing to subtract to it, so the result of the difference is null;
+ - the difference between two non-null values is their algebraic difference; or null, if the result is negative and `nonNegative: true`;
+ - null minus some value is always null;
+ - some value `v` minus null is `v` minus the last non-null value seen before `v`; or null if `v` is the first non-null value seen.
+
+Example of difference:
+
+    |   A  |   B  |   C  |
+    |:----:|:----:|:----:|
```

Comment:
This table doesn't seem to render like the others in this document.

Reply:
yep, you are right. Fixed